src/lib/storage/fsst_segment/fsst_segment_iterable.hpp

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <type_traits>
+
+#include "storage/segment_iterables.hpp"
+
+#include "storage/fsst_segment.hpp"
+#include "storage/vector_compression/resolve_compressed_vector_type.hpp"
+
+namespace opossum {
+
+template <typename T>
+class FSSTSegmentIterable : public PointAccessibleSegmentIterable<FSSTSegmentIterable<T>> {
+ public:
+  using ValueType = T;
+
+  explicit FSSTSegmentIterable(const FSSTSegment<T>& segment) : _segment{segment} {}
+
+  template <typename Functor>
+  void _on_with_iterators(const Functor& functor) const {
```

Comment:
Again public methods with "_" prefix.

Reply:
We did it the way it was implemented in the other segments.

Dictionary:
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/dictionary_segment/dictionary_segment_iterable.hpp#L25

Frame of reference:
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/frame_of_reference_segment/frame_of_reference_segment_iterable.hpp#L20

LZ4:
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/lz4_segment/lz4_segment_iterable.hpp#L20

Reply:
Yes, that's Hyrise legacy. :)